src/server/server.ts

```diff
@@ -200,6 +259,93 @@ registerRoute('post', '/compile/snippet', compileCommon);
  */
 registerRoute('post', '/compile/page', (req, res) => compileCommon(req, res, true /*wrapWithRunnerChrome*/));
 
+registerRoute('get', '/open-in-playground/:correlationId/:host/:type/:id/:filename', async (req, res) => {
+    let relativePath, templateName;
+    switch (req.params.host.toUpperCase()) {
+        case 'EXCEL':
+            relativePath = 'xl';
+            templateName = 'excel-template';
+            break;
+        case 'WORD':
+            relativePath = 'word';
+            templateName = 'word-template';
+            break;
+        case 'POWERPOINT':
+            relativePath = 'ppt';
+            templateName = 'powerpoint-template';
+            break;
+        default:
+            throw new Error(`Unsupported host: ${req.params.host}`);
+    }
+
+    const correlationId = req.params.correlationId;
+    let directoryInfo = generatedDirectories[correlationId];
+    // Check if file already exists on server for correlation id
+    if (directoryInfo) {
+        directoryInfo.isBeingRead = true;
+        fs.createReadStream(directoryInfo.name).pipe(res);
+        res.attachment(req.params.filename);
+        res.on('finish', () => {
```

Comment:
Can you open a backlog item in github for adding error cases? I don't think they are super important, but the server will just timeout if we fail to read/write any of these file operations.

Reply:
Will add.